gear-programs/vft-treasury/src/services/mod.rs

```diff
@@ -0,0 +1,452 @@
+    }
+}
+
+#[derive(Encode, Decode, TypeInfo)]
+pub enum VftTreasuryEvents {
+    Deposit {
+        from: ActorId,
+        to: H160,
+        token: ActorId,
+        amount: U256,
+    },
+    Withdraw {
+        receiver: ActorId,
+
+        token: ActorId,
+        amount: U256,
+    },
+}
+
+#[service(events = VftTreasuryEvents)]
+impl<T> VftTreasury<T>
+where
+    T: ExecContext,
+{
+    pub fn ensure_admin(&mut self, admin: ActorId) -> Result<(), Error> {
+        if self.data().admin != self.exec_context.actor_id() {
+            return Err(Error::NotAdmin);
+        }
+
+        self.data_mut().admin = admin;
+        Ok(())
+    }
+
+    pub fn update_config(&mut self, config: Config) -> Result<(), Error> {
+        if self.data().admin != self.exec_context.actor_id() {
+            return Err(Error::NotAdmin);
+        }
+
+        unsafe {
+            CONFIG = Some(config);
+        }
+
+        Ok(())
+    }
+
+    pub fn map_vara_to_eth_address(
+        &mut self,
+        ethereum_token: H160,
+        vara_token: ActorId,
+    ) -> Result<(), Error> {
+        let data = self.data();
+
+        if data.admin != self.exec_context.actor_id() {
+            return Err(Error::NotAdmin);
+        }
+
+        for (vara, eth) in self.data().vara_eth_mapping.iter() {
+            if vara == &vara_token || eth == &ethereum_token {
+                return Err(Error::DuplicateAddressMapping);
+            }
+        }
+
+        self.data_mut()
+            .vara_eth_mapping
+            .push((vara_token, ethereum_token));
+
+        Ok(())
+    }
+
+    pub fn unmap_vara_to_eth_address(
+        &mut self,
+        ethereum_token: H160,
+        vara_token: ActorId,
+    ) -> Result<(), Error> {
+        let data = self.data();
+
+        if data.admin != self.exec_context.actor_id() {
+            return Err(Error::NotAdmin);
+        }
+
+        let ix = self
+            .data()
+            .vara_eth_mapping
+            .iter()
+            .enumerate()
+            .find(|(_, map)| *map == &(vara_token, ethereum_token))
+            .map(|(ix, _)| ix)
+            .ok_or(Error::NoCorrespondingEthAddress)?;
+
+        self.data_mut().vara_eth_mapping.swap_remove(ix);
+
+        Ok(())
+    }
+
+    pub fn update_ethereum_event_client_address(
+        &mut self,
+        new_address: ActorId,
+    ) -> Result<(), Error> {
+        let data = self.data();
+        if data.admin != self.exec_context.actor_id() {
+            return Err(Error::NotAdmin);
+        }
+
+        self.data_mut().ethereum_event_client = new_address;
+        Ok(())
+    }
+
+    pub fn update_bridging_payment_service_address(
+        &mut self,
+        new_address: ActorId,
+    ) -> Result<(), Error> {
+        let data = self.data();
+        if data.admin != self.exec_context.actor_id() {
+            return Err(Error::NotAdmin);
+        }
+
+        self.data_mut().bridging_payment_service = new_address;
+        Ok(())
+    }
```

Comment:
It seems this contract should not depend on `bridging_payment_service` address so it should be removed